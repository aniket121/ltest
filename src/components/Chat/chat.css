.Chat {
  position: absolute;
  right: 0;
  top: 0;
  background-color: rgba(0,0,0,0.5);
  height: 100vh;
  width: 0vw;
  max-width: 440px;
  background-color: white;
  padding: 0px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  box-sizing: border-box;
  transition: 0.3s;
  z-index: -100;
}

.docked {
  right: -1000px;
  transition: width 0.3s;
}

.undocked {
  z-index: 2;
  width: 25vw;
  padding: 24px;
  transition: width 0.3s;
}

.pictureCircle {
  height: 56px;
  width: 56px;
  border-radius: 50%;
}

.ChatMessageList {
  overflow: auto;
}

.ChatMessageWrapper {
  display: flex;
  margin-bottom: 24px;
}

.messageAvatar {
  flex: 2;
  position: relative;
}

.messageContent {
  flex: 7;
}

.messageAuthor {
  font-family: "F37 Ginger";
  font-size: 18px;
}

.messageAuthor::after {
  content: "  ";
}

.messageDate {
  font-family: "F37 Ginger";
}

.ChatInputWrapper {
  display: flex;
  padding: 16px 12px;
  height: 56px;
  margin-top: 200%;
  
}

.Input {
  flex: 4;
  font-size: 16px;
  outline: none;
  border: none;
  color: #373737;
}

.Input::placeholder {
  color: rgba(0, 0, 0, 0.3);
}

.ChatInputButtons {
  display: flex;
  align-items: center;
  flex: 1;
  color: rgba(0, 0, 0, 0.3);
}

.sendButton {
  cursor: pointer;
  display: inline-block;
  margin-right: 8px;
}

.emojiButton {
  cursor: pointer;
  display: inline-block;
}