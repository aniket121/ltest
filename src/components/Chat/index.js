import React, { PureComponent } from 'react';
import SocialIconLive from 'components/SocialIconLive';
import Paper from 'material-ui/Paper';
import moment from 'moment';
import styles from './chat.css';

export const ChatMessage = ({ 
  message, 
  name,
  picture,
  createdAt,
  provider
}) => {
  return (
    <div className={styles.ChatMessageWrapper}>
      <div className={styles.messageAvatar}>
        <img className={styles.pictureCircle} src={picture} alt="" />
        <SocialIconLive 
          provider={provider}
          styleOverrides={{
            position: 'absolute',
            bottom: 0,
            right: 10
          }}
        />
      </div>
      <div className={styles.messageContent}>
        <div className={styles.messageHeader}>
          <span className={styles.messageAuthor}>
            {name}
          </span>
          <span className={styles.messageDate}>
            {moment(createdAt).format('h:mmA')}
          </span>
        </div>
        <div className={styles.messageBody}>
          {message}
        </div>
      </div>
    </div>
  );
};

export const ChatInput = ({ messageInput = '', onChange }) => 
  <Paper className={styles.ChatInputWrapper}>
    <input 
      className={styles.Input}
      type="text"
      name="messageInput"
      placeholder="write a comment..."
      onChange={onChange}
      value={messageInput}
    />
    <div className={styles.ChatInputButtons}>
      <span className={styles.sendButton}>
        <i className="material-icons">send</i>
      </span>
      <span className={styles.emojiButton}>
        <i className="material-icons">tag_faces</i>
      </span>
    </div>
  </Paper>

export default class Chat extends PureComponent {
  componentDidUpdate(prevProps) {
    const messageCountDidChange = this.props.messages.length !== prevProps.messages.length;
    if (messageCountDidChange) {
      const chatMessageList = document.querySelector(`.${styles.ChatMessageList}`);
      chatMessageList.scrollTop = chatMessageList.scrollHeight;
    }
  }

  render() {
    let { isChatDocked, messages } = this.props;

    if (!messages) {
      messages = [];
    }


    return (
      <div className={styles.Chat + (isChatDocked ? '' : ` ${styles.undocked}` )}>
        <div className={styles.ChatMessageList}>
          {messages.map((message, i) => (
            <ChatMessage key={i} {...message} />
          ))}
          

        </div>

        
      </div>

    );
  }
}