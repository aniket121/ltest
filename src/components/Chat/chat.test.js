import React from 'react';
import { shallow } from 'enzyme';
import Chat, { 
  ChatMessage,
  ChatInput 
} from 'components/Chat';

const messages = [
  {
    author: 'Seba Pereyro',
    createdAt: new Date().toISOString(),
    message: 'This is cool?'
  },
  {
    author: 'Nick Evans',
    createdAt: new Date().toISOString(),
    message: 'Nice job mate!'
  },
  {
    author: 'Ian Wilson',
    createdAt: new Date().toISOString(),
    message: 'See you, space cowboy'
  }
];

const createWrapperProps = props => ({
  messages: [],
  ...props
})

const createProps = props => ({
  message: 'A default message',
  author: 'Unknown Jones',
  createdAt: "2017-09-01T03:06:29.980Z",
  ...props
});

const createInputProps = props => ({
  messageInput: '',
  onChange: jest.fn(),
  ...props
})

describe('Chat', () => {
  test('should render with default props', () => {
    const props = createWrapperProps();
    const wrapper = shallow(<Chat {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should render list of messages', () => {
    const props = createWrapperProps({ messages });
    const wrapper = shallow(<Chat {...props} />);
    expect(wrapper.find(ChatMessage)).toHaveLength(props.messages.length);
  });

  // test('should render message input', () => {
  //   const props = createWrapperProps()
  //   const wrapper = shallow(<Chat {...props} />);
  //   expect(wrapper.find(ChatInput)).toHaveLength(1);
  // })
});

describe('ChatMessage', () => {
  test('should render with default props', () => {
    const props = createProps();
    const wrapper = shallow(<ChatMessage {...props} />);
    expect(wrapper).toMatchSnapshot();
  })
});

// describe('ChatInput', () => {
//   test('should render with default props', () => {
//     const props = createInputProps();
//     const wrapper = shallow(<ChatInput {...props} />);
//     expect(wrapper).toMatchSnapshot();
//   })

//   test('should render input buttons', () => {
//     const props = createInputProps();
//     const wrapper = shallow(<ChatInput {...props} />);
//     expect(wrapper.find('i.material-icons')).toHaveLength(2);
//   })
// });