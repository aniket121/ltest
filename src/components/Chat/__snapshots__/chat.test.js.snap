exports[`Chat should render with default props 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="undefined">
    <div
        className={undefined}>
        
    </div>
</div>,
  "nodes": Array [
    <div
      className="undefined">
      <div
            className={undefined}>
            
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <Chat
        messages={Array []} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Chat {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "messages": Array [],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="undefined">
          <div
                    className={undefined}>
                    
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="undefined">
          <div
                    className={undefined}>
                    
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Chat
    messages={Array []} />,
}
`;

exports[`ChatMessage should render with default props 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className={undefined}>
    <div
        className={undefined}>
        <img
            alt=""
            className={undefined}
            src={undefined} />
        <SocialIconLive
            provider={undefined}
            styleOverrides={
                Object {
                    "bottom": 0,
                    "position": "absolute",
                    "right": 10,
                  }
            } />
    </div>
    <div
        className={undefined}>
        <div
            className={undefined}>
            <span
                className={undefined} />
            <span
                className={undefined}>
                8:06PM
            </span>
        </div>
        <div
            className={undefined}>
            A default message
        </div>
    </div>
</div>,
  "nodes": Array [
    <div
      className={undefined}>
      <div
            className={undefined}>
            <img
                  alt=""
                  className={undefined}
                  src={undefined} />
            <SocialIconLive
                  provider={undefined}
                  styleOverrides={
                        Object {
                              "bottom": 0,
                              "position": "absolute",
                              "right": 10,
                            }
                  } />
      </div>
      <div
            className={undefined}>
            <div
                  className={undefined}>
                  <span
                        className={undefined} />
                  <span
                        className={undefined}>
                        8:06PM
                  </span>
            </div>
            <div
                  className={undefined}>
                  A default message
            </div>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Unknown
        author="Unknown Jones"
        createdAt="2017-09-01T03:06:29.980Z"
        message="A default message" />,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "author": "Unknown Jones",
          "createdAt": "2017-09-01T03:06:29.980Z",
          "message": "A default message",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 3,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className={undefined}>
          <div
                    className={undefined}>
                    <img
                              alt=""
                              className={undefined}
                              src={undefined} />
                    <SocialIconLive
                              provider={undefined}
                              styleOverrides={
                                        Object {
                                                  "bottom": 0,
                                                  "position": "absolute",
                                                  "right": 10,
                                                }
                              } />
          </div>
          <div
                    className={undefined}>
                    <div
                              className={undefined}>
                              <span
                                        className={undefined} />
                              <span
                                        className={undefined}>
                                        8:06PM
                              </span>
                    </div>
                    <div
                              className={undefined}>
                              A default message
                    </div>
          </div>
</div>,
        "_debugID": 6,
        "_renderedOutput": <div
          className={undefined}>
          <div
                    className={undefined}>
                    <img
                              alt=""
                              className={undefined}
                              src={undefined} />
                    <SocialIconLive
                              provider={undefined}
                              styleOverrides={
                                        Object {
                                                  "bottom": 0,
                                                  "position": "absolute",
                                                  "right": 10,
                                                }
                              } />
          </div>
          <div
                    className={undefined}>
                    <div
                              className={undefined}>
                              <span
                                        className={undefined} />
                              <span
                                        className={undefined}>
                                        8:06PM
                              </span>
                    </div>
                    <div
                              className={undefined}>
                              A default message
                    </div>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Unknown
    author="Unknown Jones"
    createdAt="2017-09-01T03:06:29.980Z"
    message="A default message" />,
}
`;
